v2/pkg/genruntime/configmaps/validation.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package configmaps
+
+import (
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+type keyPair struct {
```

Comment:
Should we consider having a common `keyPair` struct to share between both `configmaps/validation` and `secrets/validation`?

Reply:
Given it's just a small helper it didn't feel valuable, but happy to do so if the amount of duplication grows.